projecttacoma/fqm-testify
# Resource code and display name added

## Summary
Display the code and the first few words of the display text next to the test case resources in order to differentiate between resources that are of the same type. 

## New Behavior
When the display text is too long for the page, the text is cut off by an ellipses. To see the full text, the use can hover over the text with their mouse and the full text will appear in a popup "tooltip". 

## Code Changes
- `fhir.ts`: function added that takes a `Fhir4Resource` as a parameter and returns a string of that resource's code and display text. 
- `TestResourceCreation`: the `getFhirResourceString` in `fhir.ts` is called in order to display the code and display text next to the resourceType. In addition, this text is wrapped in a tooltip component so that the user can hover over the text to see the full text if it is cut off.
- `TestResourceCreation.test.tsx`: unit tests were added to test these changes.
- fhirpath added as a dependency.

## Testing Guidance
Run unit tests with `npm run test`. Then, try the following in the app:
1. Upload a measure bundle JSON file. 
2. Create a test patient.
3. Add some test case resources to that patient. The test case resources should display "ResourceType (code: display)".
4. Click the Edit button. Make sure that the values in the JSON for code and display match the ones shown. 
5. Add test case resources until you get one where the text is so long that it is cut off on the page with ellipses. 
6. Hover over the cut off text. You should see a popup that contains the full display text of the test case resource.
